src/rules/no-anonymous-default-export.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @fileoverview Rule to disallow anonymous default exports.
+ * @author Duncan Beevers
+ */
+
+const MAP = {
+  Literal: 'allow-literal',
+  ObjectExpression: 'allow-object-expression',
+  FunctionDeclaration: 'allow-anonymous-function-declaration',
+  ArrowFunctionExpression: 'allow-arrow-function-expression',
+  ClassDeclaration: 'allow-anonymous-class-declaration',
+}
+
+const FLAGS = []
+for (const nodeType in MAP) {
+  FLAGS.push(MAP[nodeType])
+}
+
+module.exports = {
+  meta: {
+    schema: {
+      type: 'array',
```

Comment:
I think the options should better be passed as an object, with a `true`/`false` values to each key. I found that having one object with settings instead of multiple values in a given order makes it much more extendable.

My gut feeling is to make it so that a field set to `true` would make it forbidden, and `false` would make it ignored, and that you'd have it `true` by default.

```js
/*eslint import/no-anonymous-default-export: [2, "allow-arrow-function-expression"]*/
-->
/*eslint import/no-anonymous-default-export: [2, {ArrowFunctionExpression: false}]*/
```

The schema would look something like this (IIRC) (adapted from what is done in https://github.com/benmosher/eslint-plugin-import/blob/master/src/rules/no-extraneous-dependencies.js)

```js
 schema: [
      {
        'type': 'object',
        'properties': {
          'Literal': { 'type': 'boolean' },
          'ObjectExpression': { 'type': 'boolean' },
          'FunctionDeclaration': { 'type': 'boolean' },
          'ArrowFunctionExpression': { 'type': 'boolean' },
          'ClassDeclaration': { 'type': 'boolean' },
        },
        'additionalProperties': false,
      },
    ],
```

And you'd get options like
```js

  create: function (context) {
    const options = context.options[0] || {}
  }
```